### ISSUE TITLE
Points accessor for datasets?

### ISSUE THREAD

**Issue by takluyver** [h5py/h5py#1602]
HDF5 has a way to select points in a dataset, passing coordinates for each point to [H5Sselect_elements](https://confluence.hdfgroup.org/display/HDF5/H5S_SELECT_ELEMENTS). This is exposed in the low-level API of h5py, but the only way to use it in the high level is with a mask array (an boolean array with the same shape as the dataset). This is only practical for smaller datasets where you can make this mask in memory.

I propose adding a new way to read/write individual points in a dataset, something like this:

```python
dset.points[[
    (0, 5),
    (2, 6),
    (3, 9),
]]
```

i.e. that would read 3 points from a 2D dataset. I think it's easier to understand which points we're looking at this way than with numpy fancy indexing on multiple dimensions, where the equivalent would look like `arr[[0, 2, 3], [5, 6, 9]]`.

This may be something for after 3.0.

**Comment by dota17** [comment]
Yes, I think this is more easier to understander these fancy indexing operation.
